src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,151 @@
+
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%" and "set %types%'[s] property to %boolean%"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void register(Class<? extends SetEffect<?>> effect, String property, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%");
+	}
+
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%", "set %types%'[s] property to %boolean%"
+	 * and "make %types% makeProperty" with "force %types% to makeProperty"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param makeProperty The property of the syntax used for "make %types% makeProperty".
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void registerMake(Class<? extends SetEffect<?>> effect, String property, String makeProperty, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + makeProperty,
+				"force %" + type + "% to " + makeProperty,
+				"make %" + type + "% not " + makeProperty,
+				"force %" + type + "% (to not|not to) " + makeProperty);
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		make = matchedPattern >= 2;
+		if (exprs.length == 1) {
+			if (!make)
+				throw new SkriptAPIException("There was not two expressions in the SetEffect class '" + getClass().getName() + "' exprs: " + Arrays.toString(exprs));
+			expression = (Expression<T>) exprs[0];
+			negated = matchedPattern >= 4;
+			return true;
+		}
+		expression = (Expression<T>) exprs[matchedPattern];
+		value = (Expression<Boolean>) exprs[matchedPattern ^ 1];
+		return true;
+	}
+
+	protected final boolean isNegated() {
+		return negated;
+	}
+
+	/**
+	 * Returns the expression that was registered as the type for this SetEffect.
+	 * 
+	 * @return Expression<T> for getting the values of the expression used.
+	 */
+	protected final Expression<T> getExpression() {
+		return expression;
+	}
+
+	/**
+	 * For usage when overriding the {@link #init(Expression[], int, Kleenean, ParseResult)} method.
+	 * Set the expression to be used for the SetEffect utility class.
+	 * 
+	 * @param expression The expression that contains the data value for the SetEffect.
+	 */
+	protected final void setExpression(Expression<T> expression) {
+		this.expression = expression;
+	}
+
+	/**
+	 * Return a BiComsumer that will be used to apply the boolean value to the type.
+	 */
+	protected abstract BiConsumer<T, Boolean> apply();
```

Comment:
I'm still not sure I really see the point of the BiConsumer - can't we just pass these as parameters?

Reply:
This is in the modern times, we have Java 8, it's nice to have lambdas now.

Reply:
It just kind of seems like an unnecessary layer to me. I welcome discussion here.

Reply:
agreed with pickle on this one